### PR TITLE
Teleport sync folders

### PR DESCRIPTION
**Pull request by fiskus**
Frontend for managing local folder - s3 folder pairs

TODO:
* add workflow
* probably default meta

Screenshot from 2022-04-27 12-05-22

### CODE REVIEW

File: catalog/app/containers/Admin/Sync.tsx [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+  const handleClick = React.useCallback(async () => {
+    const newLocalPath = await ipc.invoke(IPC.EVENTS.LOCALPATH_REQUEST)
+    if (!newLocalPath) return
+    onChange(newLocalPath)
+  }, [ipc, onChange])
+
+  return (
+    <Form.Field
+      disabled={false}
```
Comment (nl0) [inline catalog/app/containers/Admin/Sync.tsx:31]: is this necessary?

File: catalog/app/containers/Admin/Sync.tsx [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      id="localPath"
```
Comment (nl0) [inline catalog/app/containers/Admin/Sync.tsx:32]: i don't think short ids are safe bc of a possibility of collisions. what's its purpose?

File: catalog/app/containers/Admin/Sync.tsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+                placeholder="s3://bucket/namespace/package"
+                validate={
+                  validators.composeAnd(
+                    validators.required,
+                    validators.s3Url,
+                  ) as FF.FieldValidator<any>
+                }
+                errors={{
+                  required: 'S3 URL is required',
+                  s3Url: 'Enter valid S3 url to package',
```
Comment (nl0) [inline catalog/app/containers/Admin/Sync.tsx:134]:
```suggestion
                  s3Url: 'Enter valid S3 URL to package',
```
this is not technically correct, but whatever, it makes sense in this context

**Review by nl0 — APPROVED** [review]
looks ok

File: catalog/app/containers/Admin/Sync.tsx (hunk 2 above)
Comment (fiskus) [reply catalog/app/containers/Admin/Sync.tsx]: This code does nothing too

File: catalog/app/containers/Admin/Sync.tsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+  const [folders, setFolders] = React.useState<null | DataRow[]>(null)
+  React.useEffect(() => {
+    async function fetchData() {
+      const config = await ipc.invoke(IPC.EVENTS.SYNC_FOLDERS_LIST)
+      setFolders(config.folders)
+    }
+
+    fetchData()
+  }, [ipc, key])
+  return folders ? [folders, inc] : [null, inc]
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/containers/Admin/Sync.tsx]: You right, thanks

File: catalog/app/containers/Admin/Sync.tsx (hunk 1 above)
Comment (fiskus) [reply catalog/app/containers/Admin/Sync.tsx]: Fix https://github.com/quiltdata/quilt/pull/2829

**Pull request merged**